microsoft/botframework-solutions
# Updates to Virtual Assistant Android app

## Description
- Close #1439: Make Adaptive card selections work by tapping.
- Close #1440: Show and make clickable suggested actions
- Cleaned up comments
- Long Adaptive Card JSON is fully echoed to logcat (no longer cut-off prematurely)
- Fixed race-condition; speechSdk can be null when GPS location is quickly obtained
- Added 15sec Bot response timeout so that app doesn't wait forever when a response is not sent
- Fixed Location and Timezone event names

### Architecture

### Bug Fixes
fixes bug where played audio samples weren't released

### Conversational Experience

### Maintenance

### Language Understanding

## Testing Steps
Ask the bot "find a coffee shop". When it proposes a location say "Yes". A carousel of adaptive cards of coffee shops is shown. Touch one of the cards. Observe that the bot asks whether to show directions.
Suggested Actions isn't quite testable yet because the server isn't passing that data to the app yet - the feature was smoke-tested using the following JSON https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-add-suggested-actions?view=azure-bot-service-4.0

## Checklist

- [X ] I have commented my code, particularly in hard-to-understand areas
